src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisValidateOnlyOverlappingFalse.java

```diff
@@ -478,3 +478,12 @@ private void hashCodeMethod(String param) {
         hashCode();
     }
 }
+class Issue7306 {
+    java.util.List<String> add = new java.util.ArrayList<>();
+
+    void someMethod() {
+        java.util.List<String> test = new java.util.ArrayList<>();
+        test.forEach(this.add::add); // OK
+        test.forEach(add::add); // violation
```

Comment:
Please add test for reference by class name https://strkkk.github.io/checkstyle/7306/Vavr/xref/checkstyle/contribution/checkstyle-tester/repositories/Vavr/vavr/src/main/java/io/vavr/collection/HashArrayMappedTrie.java.html#L473

And https://strkkk.github.io/checkstyle/7306/guava/xref/checkstyle/contribution/checkstyle-tester/repositories/guava/guava-testlib/src/com/google/common/collect/testing/AbstractIteratorTester.java.html#L330 where no `this` and all variables are local to method.

We need to copy all interesting cases from regression report, to less depends on it. In some ideal future, execution on our Inputs should cover all cases/combination of code.

Reply:
done